### ISSUE TITLE
Import data 2.2.6 Value for 'product_type' attribute contains incorrect value

### ISSUE THREAD

**Issue by ankit-sf** [magento/magento2#19346]
### Preconditions (*)

1. magento 2.2.6 (vanilla instance)

### Steps to reproduce (*)

1. create an attribute with code product_type http://i.prntscr.com/AwUJuhizQKatBwNwRa1S7Q.png with type: dropdown scope: global sortable, comparable: no
2. add it to a family and create a product.
3. go to import profiles download sample file then reimport sample file

### Expected result (*)

1. [Screenshots, logs or description]
2. import should be completed

### Actual result (*)

1.  Value for 'product_type' attribute contains incorrect value, see acceptable values on settings specified for Admin in row(s): 1, 2, 3, 4, 5, 6
http://i.prntscr.com/lsI-FNJnTzO0gEttdZp1wA.png

2. Also reproduced on earlier versions.

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@ankit-sf do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by ankit-sf** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf, here is your [Magento instance](https://i-19346-2-2-6.instances.magento-community.engineering).
Admin access: [https://i-19346-2-2-6.instances.magento-community.engineering/admin](https://i-19346-2-2-6.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ankit-sf** [comment]
yes, i confirm that i was able to reproduce the issue on vanilla Magento instance following steps to reproduce

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @ankit-sf  why you think this is a bug? if you add some aditional atributes you need to add this to your csv file, so sample csv is not working any more, if you add some additional attributes in default attribute set. this is not a bug.

**Comment by ankit-sf** [comment]
no that's not the case. because newly added custom attributes is **not required** field.
the issue is that after adding a field, any CSV import doesn't work without that attribute and there is an error.

**Comment by ghost** [comment]
@ankit-sf This what i'm talking about, you need to use your custom attribute set, not default, you set **VALUE** is not required, but not an attribute, if you add this to you custom attribute set you dont have an error.

**Comment by ghost** [comment]
@ankit-sf in sample csv file
![selection_217](https://user-images.githubusercontent.com/40756439/48937580-3d81c080-ef17-11e8-9017-2eaafab5afc6.png)

**Comment by ankit-sf** [comment]
@engcom-backlog-nazar  i'll check and revert in some minutes

**Comment by ankit-sf** [comment]
@engcom-backlog-nazar Issue exists
Although it's clear in description above, i am rephrasing issue.
### Case 1:  no issue with code other than product_type
step 1: on vanilla m2 instance, create attribute (dropdown, not required) with **code like test_select**
step2: add it in default attribute set
3. goto Import download sample file reimport it.
Result: everything works fine

### Case 2: Issue with code product_type
step 1: on vanilla m2 instance, create attribute (dropdown, not required) with **code product_type**
step2: add it in default attribute set
3. goto Import download sample file reimport it.
Result: Error message shows  Value for 'product_type' attribute contains incorrect value

**Comment by ghost** [comment]
@ankit-sf thanks now clear, this issue occurs only with attribute code product_type with dropdown.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96611`, `MAGETWO-96612` were created

**Comment by magento-engcom-team** [comment]
Hi @hardik-krish. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by hardik-krish** [comment]
Hi @ankit-sf,
Magento itself consider product_type as reserve keyword if i am correct.
If you want to see in detail then export  product from admin  and check CSV file.
you will find there is product type column is already there.
So whenever some one will try to create attribute using product_type code it should not allowed to create  attribute and it should be display message like this is reserve keyword.

**Comment by ankit-sf** [comment]
@hardik-krish maybe you didn't correctly checked issue.
i know there are product_type reserved keyword that is causing issue.
1. obviously there can solution to implement so that with product_type keyword attributes could not be created.
but if that's case then you should have created another issue on github with this information. then you should have closed this issue.
I hope you understand my concern

2nd solution would be to correct it by change in code so correct seperate validation is applied on reserved product_type and custom attribute product_type.

I think you should reopen this issue, because issue exist and also there is no issue corresponding to solution you mentioned.

**Comment by ghost** [comment]
@hardik-krish @ankit-sf is right there must be validation of the available attributes

**Comment by ankit-sf** [comment]
@engcom-backlog-nazar @hardik-krish 
thank you for referencing related PR. that clears it now.

Now, I have just one minor concern,
we have migrated from magento1 to magento2 have an attribute with product_type code. we have options in attribute and lot of products having this attribute.
since the current solution is to disallow attribute_code with product_type.
what implementation we should do on magento2 when updating magento2 to latest patch so that our data is relevant.
these solutions i can think of:
1. change code of attribute, but by browser UI it is disabled, so maybe change by db. (but can break things)
2. delete attribute and create another one with same data and reassign in products (lot of work since lot of products)

so I just have a suggestion (it may not be relevant)
Instead of disallowing product_type code in the attribute.
correct it by the change in code so correct separate validation is applied on reserved product_type and custom attribute product_type. may be i can help with code and do a PR. this way it would not break data of customers already having product_type attribute code.

**Comment by orlangur** [comment]
@ankit-sf,

> correct seperate validation is applied on reserved product_type and custom attribute product_type

This totally does not make sense.

What kind of problem are you trying to solve by attribute named `product_type`? It is obviously quite confusing to have such attribute taking into account "product type" already has special meaning in Magento.

**Comment by ankit-sf** [comment]
ok then disallow product_type is the best option. thanks.

**Comment by GovindaSharma** [comment]
@ankit-sf i have done the same thing..i disallowed that attribute to be created in magento.to disallow that   attribute being created will be the best option.

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf. Thank you for your report. 
The issue has been fixed in magento/magento2#19408 by @GovindaSharma in 2.3-develop branch
Related commit(s):
 - [ca0da34ee2691bfaca1c74d94a0db33377d4e54c](https://github.com/magento/magento2/commit/ca0da34ee2691bfaca1c74d94a0db33377d4e54c)

The fix will be available with the upcoming 2.3.1 release.

**Comment by maxq** [comment]
Still seeing this in 2.4.5 where is the fix?

**Comment by usuyukiso** [comment]
reproduced on ver.2.4.6-p6
